src/common/getType.ts

```diff
@@ -356,14 +369,23 @@ function getOpCodeType(
     case "bit_shift_left":
     case "bit_shift_right":
     case "min":
-    case "max":
-      expectType(integerType(), integerType());
-      return getArithmeticType(
-        expr.op,
-        types[0] as IntegerType,
-        types[1] as IntegerType,
-        expr.source
-      );
+    case "max": {
+      const op = expr.op;
+      if (isAssociative(expr.op)) {
```

Comment:
The list of associative ops is fixed. Splitting the block of cases into two seems more natural than having an isAssociative check. I feel I'm missing something.